play-json/src/test/scala/play/api/libs/json/MacroSpec.scala

```diff
@@ -77,6 +120,34 @@ class MacroSpec extends org.specs2.mutable.Specification {
         "ipsum" -> 2.34F, "age" -> 2
       )
     }
+
+    "be generated for a sealed family" >> {
+      implicit val simpleWrites = Writes[Simple] { simple =>
+        Json.obj("bar" -> simple.bar)
+      }
+      implicit val optionalWrites: OWrites[Optional] = Json.writes[Optional]
+      implicit val familyWrites: OWrites[Family] = Json.writes[Family]
+
+      val simple = Simple("foo")
+      val optional = Optional(None)
+
+      val jsSimple = simpleWrites.writes(simple).asInstanceOf[JsObject] + (
+        "_type" -> JsString("play.api.libs.json.MacroSpec.Simple")
+      )
+
+      val jsOptional = optionalWrites.writes(optional) + (
+        "_type" -> JsString("play.api.libs.json.MacroSpec.Optional")
+      )
+
+      lazy val wsimple = Json.toJson[Family](simple)
+      lazy val wopt = Json.toJson[Family](optional)
+
+      wsimple must_== jsSimple and {
+        wsimple.validate(Json.reads[Simple]).get must beTypedEqualTo(simple)
+      } and (wopt must_== jsOptional) and {
+        wopt.validate(Json.reads[Optional]).get must beTypedEqualTo(optional)
```

Comment:
@gmethvin Updated to have the `_type` discriminator at the same level as the other properties